### PR TITLE
Fix elementwise_pow bug on CUDA place with integer

### PR DESCRIPTION
**Pull request by zhiqiu**
On CUDA place, `elementwise_pow` fails when input integers, for example,
image

image

This PR fixes that.

### CODE REVIEW

File: paddle/fluid/operators/elementwise/elementwise_pow_op.h [hunk 1]
```diff
@@ -12,17 +12,59 @@ limitations under the License. */
...
 namespace operators {
 
-template <typename T>
-struct PowFunctor {
+namespace details {
+
+// Only valid if T is an integral type.
+template <class T,
+          class = typename std::enable_if<std::is_integral<T>::value>::type>
+static inline HOSTDEVICE T ipow(T base, T exp) {
```
Comment (sneaxiy) [inline paddle/fluid/operators/elementwise/elementwise_pow_op.h]: `static_assert(std::is_integral<T>::type, "T must be integral");`, and remove `std::enable_if`?

File: paddle/fluid/operators/elementwise/elementwise_pow_op.h [hunk 2]
```diff
@@ -12,17 +12,59 @@ limitations under the License. */
...
+  if (exp < 0) {
+    // Note(zhiqiu): In some library, like numpy and pytorch,
+    // integers to negative integer powers are not allowed.
+    // Since a^b = 1/(a^(-b)) when b < 0, it will truncate to 0.
+    // Return 0, for specialization to avoiding infinite loop.
+    return 0;
+  }
+
+  T result = 1;
```
Comment (sneaxiy) [inline paddle/fluid/operators/elementwise/elementwise_pow_op.h]: Add a link where the following codes from?

File: paddle/fluid/operators/elementwise/elementwise_pow_op.h (hunk 1 above)
Comment (zhiqiu) [reply paddle/fluid/operators/elementwise/elementwise_pow_op.h]: Thanks, implementation changed.

File: paddle/fluid/operators/elementwise/elementwise_pow_op.h (hunk 2 above)
Comment (zhiqiu) [reply paddle/fluid/operators/elementwise/elementwise_pow_op.h]: Thanks, implementation changed.

**Review by sneaxiy — APPROVED** [review]
LGTM.

**Pull request merged**